### PR TITLE
new interactions system

### PR DESCRIPTION
**Pull request by Nyameliaaaa**
merges the related elements of an interaction into one file (avatar component, context menu and chat input command in one file for example)

### CODE REVIEW

**Comment by hkubus** [comment]
Ok ?

**Comment by Nyameliaaaa** [comment]
nvm

**Review by SuperchupuDev — APPROVED** [review]

File: src/classes/Client.ts [hunk 1]
```diff
@@ -3,26 +3,30 @@ import process from 'node:process';
...
 export class FluorineClient extends Client {
     createdAt = performance.now();
     logger = Logger;
     i18n = i18next;
     prisma = new PrismaClient({});
 
-    commands = new CommandHandler(this);
-    components = new ComponentHandler(this);
-    modals = new ModalHandler(this);
+    interactions = new CommandHandler(this);
```
Comment (Nyameliaaaa) [inline src/classes/Client.ts]: should rename this to match

**Comment by Nyameliaaaa** [comment]
ok so i have two little minor changes to make, ill do them and then we'll see.

File: src/commands/info.ts [hunk 2]
```diff
@@ -1,16 +1,137 @@
...
+    type ChatInputCommandInteraction,
+    type Interaction,
+    type ButtonInteraction,
+    SlashCommandBuilder,
+    ActionRowBuilder,
+    ButtonBuilder,
+    ButtonStyle
+} from 'discord.js';
+
+async function getEmbed(client: FluorineClient, interaction: Interaction, page: string) {
```
Comment (Nyameliaaaa) [inline src/commands/info.ts]: can we get rid of the getembed and getcomponents functions (have the embed/component in oninteraction)?

**Pull request merged**